api/v3/Contact/Addidentity.php

```diff
@@ -23,13 +23,23 @@ function civicrm_api3_contact_addidentity($params) {
 
   // TODO: check if identifier_type exists?
 
-  // create id-entry
-  $query = CRM_Core_DAO::executeQuery(CRM_Identitytracker_Configuration::getInsertSQL(), array(
-    1 => array($params['contact_id'], 'Integer'),
-    2 => array($params['identifier_type'], 'String'),
-    3 => array($params['identifier'], 'String'),
-    4 => array($params['used_since'], 'String'),
-  ));
+  // Insert/Update using APIv4, unique tuples of contact_id, identifier_type,
+  // identifier, and possibly context.
+  \Civi\Api4\CustomValue::save(CRM_Identitytracker_Configuration::GROUP_NAME, FALSE)
```

Comment:
I think, since the current approach executed SQL on the custom group table, calling the `Contact::update()` API would be too different an approach, as this would imply invoking pre/post hooks on more entities. So calling `CustomValue::save()` should be less intrusive. But you're right, semantically this is about updating contacts.